content/post/manage-redis-secret-engine-using-kubevault-in-aws/index.md

```diff
@@ -0,0 +1,587 @@
+  --set kubedb-ops-manager.enabled=true \
+  --set kubedb-autoscaler.enabled=true \
+  --set kubedb-dashboard.enabled=true \
+  --set kubedb-schema-manager.enabled=true \
+  --set-file global.license=/path/to/the/license.txt
+```
+
+Let's verify the installation:
+
+```bash
+$ watch kubectl get pods --all-namespaces -l "app.kubernetes.io/instance=kubedb"
+
+NAMESPACE   NAME                                            READY   STATUS    RESTARTS   AGE
+kubedb      kubedb-kubedb-autoscaler-54fcfbcdb7-28wc7       1/1     Running   0          2m22s
+kubedb      kubedb-kubedb-dashboard-599fdd8d8f-tnj6l        1/1     Running   0          2m22s
+kubedb      kubedb-kubedb-ops-manager-868d684c84-7v9dz      1/1     Running   0          2m22s
+kubedb      kubedb-kubedb-provisioner-75b544fdf4-w6z8l      1/1     Running   0          2m22s
+kubedb      kubedb-kubedb-schema-manager-66d686b986-kskfj   1/1     Running   0          2m22s
+kubedb      kubedb-kubedb-webhook-server-677b88d9bb-29qln   1/1     Running   0          2m22s
+```
+
+## Install KubeVault Enterprise Operator
+
+```bash
+$ helm install kubevault appscode/kubevault \
+  --version v2022.12.28 \
+  --namespace kubevault --create-namespace \
+  --set-file global.license=/path/to/the/license.txt
+```
+Let's verify the installation:
+
+```bash
+$ watch kubectl get pods --all-namespaces -l "app.kubernetes.io/instance=kubevault"
+
+NAMESPACE   NAME                                                  READY   STATUS    RESTARTS   AGE
+kubevault   kubevault-kubevault-operator-86b8c7f688-6ln65         1/1     Running   0          22s
+kubevault   kubevault-kubevault-webhook-server-8554c7cd7f-j9w6q   1/1     Running   0          22s
+```
+#### Install KubeVault CLI
+
+KubeVault provides a `kubectl` plugin to interact with KubeVault resources. Let's install [KubeVault CLI](https://kubevault.com/docs/latest/setup/install/kubectl_plugin/)
+
+#### Install Secret-store CSI Driver
+
+```bash
+$ helm repo add secrets-store-csi-driver https://kubernetes-sigs.github.io/secrets-store-csi-driver/charts
+$ helm install csi-secrets-store secrets-store-csi-driver/secrets-store-csi-driver --namespace kube-system
+```
+#### Install Vault specific CSI Provider
+
+```bash
+$ helm repo add hashicorp https://helm.releases.hashicorp.com
+$ helm install vault hashicorp/vault \
+    --set "server.enabled=false" \
+    --set "injector.enabled=false" \
+    --set "csi.enabled=true"
+```
+
+### Create Namespace and Secret
+
+To keep everything isolated, we are going to use a separate namespace `demo` throughout this tutorial.
+
+```bash
+$ kubectl create namespace demo
+namespace/demo created
+```
+
+Here, We need to create a storage secret for our backend.
+
+```bash
+$ echo -n '<your-secret-key-id-here>' > secret_key
+$ echo -n '<your-secret-access-key-here>' > access_key
+$ kubectl create secret generic -n demo aws-secret \
+    --from-file=./secret_key \
+    --from-file=./access_key
+secret/aws-secret created
+```
+
```

Comment:
We need to create an s3 bucket in EKS to use as Vault Backend. For this example, we have created a bucket named `vault-demo-1` in `us-east-1` region.